tests/bluetooth/bsim_bt/bsim_test_adv_resume/src/peripheral.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bs_bt_utils.h"
+#include "zephyr/bluetooth/addr.h"
+#include "zephyr/bluetooth/bluetooth.h"
+#include "zephyr/bluetooth/conn.h"
+#include "zephyr/toolchain/gcc.h"
+
+#include <stdint.h>
+#include <string.h>
+
+void peripheral_procedure(void)
+{
+	bs_bt_utils_setup();
+
+	printk("DUT start\n");
+	k_msleep(1000);
+
+	/* start scanning (using NRPA) */
+	scan_connect_to_first_result();
+	k_msleep(100);
```

Comment:
Why is this sleep needed?

Reply:
so the stack has time to fully start up the scanner. Since the stack is a bit of a mess with all the different threads (and the wq) and whatnot, I prefer to give myself some room to be sure it has properly started before attempting to start the advertiser.

Reply:
Starting scanning happens in the same thread. Once e.g. `bt_le_scan_start` returns you know that the device is in a scanning state. 

Generally I highly dislike using timers to ensure state correctness.

Reply:
so on real devices, I agree 100%. In bsim tests, which are deterministic (and where waiting is instant in CPU time) afaik, it doesn't matter as much IMO.
I tried it without the sleep, and it works as you said now. But I don't think the API gives any guarantees about the operation being synchronous, so the controller may decide to do it async in the future.

Reply:
> But I don't think the API gives any guarantees about the operation being synchronous, so the controller may decide to do it async in the future.

That is true. I don't believe our APIs actually say whether they are synchronous or asynchronous operations. It is, however, the host and not controller that makes e.g. `scan_start` sync by using `bt_hci_cmd_send_sync`. @jhedberg @alwa-nordic is this something we should describe in the API in some way?